src/utils/handleSubmit.js

```diff
@@ -0,0 +1,43 @@
+import Popup from "../components/Popup.js";
+
+export function handleSubmit(
+  submitAction,
+  popupInstance, // This should be an instance of Popup or PopupDeleteCard
+  submitText = "Save",
+  loadingText = "Saving..."
+) {
+  // Check if popupInstance is an instance of Popup
+  if (!(popupInstance instanceof Popup)) {
+    console.error("The provided popupInstance is not an instance of Popup.");
+    return Promise.reject("Invalid popup instance");
+  }
+
+  const popup = popupInstance._popupElement;
+  const submitButton = popup.querySelector(".modal__save-button");
```

Comment:
You have `_submitButtonText` inside `PopupWithForm` as the submit button. You don't need to search it